src/Tools/BuildValidator/CompilationDiff.cs

```diff
@@ -114,124 +131,183 @@ private CompilationDiff(ImmutableArray<Diagnostic> diagnostics, string originalP
-                        var originalPdbMdvPath = Path.Combine(originalPath, assemblyName + ".pdb.mdv");
-                        writeVisualization(originalPeMdvPath, optionsReader.PeReader.GetMetadataReader());
-                        writeVisualization(originalPdbMdvPath, optionsReader.PdbReader);
+                    File.WriteAllText(Path.Combine(assemblyDebugPath, scriptName), $@"code --diff (Join-Path $PSScriptRoot ""{originalFilePath}"") (Join-Path $PSScriptRoot ""{rebuildFilePath}"")");
+                    string getRelativePath(string dataFilePath) => dataFilePath.Substring(assemblyDebugPath.Length);
+                }
 
-                        var originalPdbXmlPath = Path.Combine(originalPath, assemblyName + ".pdb.xml");
-                        using var originalPdbXml = File.Create(originalPdbXmlPath);
+                var sourcesPath = Path.Combine(assemblyDebugPath, "sources");
+                Directory.CreateDirectory(sourcesPath);
 
-                        var rebuildPdbXmlPath = Path.Combine(rebuildPath, assemblyName + ".pdb.xml");
+                // TODO: output source files should include the entire relative path instead of just the file name.
+                foreach (var tree in compilation.SyntaxTrees)
+                {
+                    var sourceFilePath = Path.Combine(sourcesPath, Path.GetFileName(tree.FilePath));
+                    using var file = File.OpenWrite(sourceFilePath);
+                    var writer = new StreamWriter(file);
+                    tree.GetText().Write(writer);
+                    writer.Flush();
+                }
+            }
 
-                        var pdbToXmlOptions = PdbToXmlOptions.ResolveTokens
-                            | PdbToXmlOptions.ThrowOnError
-                            | PdbToXmlOptions.ExcludeScopes
-                            | PdbToXmlOptions.IncludeSourceServerInformation
-                            | PdbToXmlOptions.IncludeEmbeddedSources
-                            | PdbToXmlOptions.IncludeTokens
-                            | PdbToXmlOptions.IncludeMethodSpans;
+            static BuildDataFiles createBuildArtifacts(string outputPath, string assemblyFileName, BuildInfo buildInfo)
+            {
+                var assemblyName = Path.GetFileNameWithoutExtension(assemblyFileName);
+                var assemblyFilePath = Path.Combine(outputPath, assemblyFileName);
+                var buildDataFiles = new BuildDataFiles(
+                    AssemblyMdvFilePath: Path.Combine(outputPath, assemblyName + ".mdv"),
+                    PdbMdvFilePath: Path.Combine(outputPath, assemblyName + ".pdb.mdv"),
+                    ILFilePath: Path.Combine(outputPath, assemblyName + ".il"),
+                    PdbXmlFilePath: Path.Combine(outputPath, assemblyName + ".pdb.xml"),
+                    CustomDataFilePath: Path.Combine(outputPath, "custom-data.txt"));
```

Comment:
should this also be prefixed with the assembly name?

Reply:
Didn't seem necessary here. It's only one per original / rebuild and the assembly name is in the grand parent directory so felt redundant to add. Don't feel strongly about this though.